assets/js/aplus.js

```diff
@@ -624,6 +625,106 @@ $.fn.highlightCode = function(options) {
 })(jQuery, window, document);
 
 
+/**
```

Comment:
Do you mean the line `$('[data-external-launch]').aplusExternalLauncher();` or the whole jQuery plugin? This JS code is used in exercise and chapter pages, as well as LTI service menu links. 

`aplus.js` has plenty of code that is only used on a subset of pages. However, it is included on all pages so that we don't need to modify individual pages to include specific JS lines.

Reply:
I meant the whole plugin.

I know that there are things in `aplus.js` that are only used by a subset of pages. However, a God file hurts understandability because it detaches the code from the context it is used in, and makes it difficult to see what is already implemented and could be reused (see e.g. the point about `getCookie` below -- its existence would have been obvious if it was in its own file `getCookie.js` instead of being buried in  `aplus.js`)

Ideally, we wouldn't have an `aplus.js` at all - the name doesn't describe the contents, which don't really have anything in common. On the long term, the file should be split into many small files with a clearly defined purpose, which then could be compiled into one minified file if wanted. Even if we are not refactoring everything right now, we should at least try to have any major new changes working towards this ideal.